多言語化(英語、日本語対応)しました

naomichi-h opened the pull request:
i18nを使用し、多言語化(英語、日本語対応)を行いました。
言語の切り替えは、urlに`?locale=ja`もしくは`?locale=en`を渡すことで行えるようにしています。
rubocop通過済です。

レビューのほど、よろしくお願いいたします。

naomichi-h committed:
add:英語、日本語対応しました

ロケール辞書(en.yml、translation_en.yml、ja.yml、translation_ja.yml)を追加しました。
ロケール辞書が参照されるように、viewファイルおよびcontrollerファイルを書き換えました。
ロケールをurlへパラメータを渡すことで変更できるようcontrollerファイルに追記しました。

maedana commented on Gemfile:
```diff
@@ -56,3 +56,5 @@ end
 
 gem 'carrierwave'
+
+gem 'i18n_generators'
```
開発時だけ必要なものだと思うので、development groupに定義されるようにしましょうー

maedana commented on app/controllers/application_controller.rb:
```diff
@@ -1,4 +1,9 @@
+  private
+
+  def default_url_options(options = {})
+    { locale: I18n.locale }.merge options
```
これだともしoptionsに`locale:`キーがあったらその値で上書きされちゃうので、`options.merge...`と順番逆がいいんじゃないかなと思いました。

naomichi-h replied:
調べたのですが、`merge`メソッドって重複するキーがある場合、引数の値で上書きされるんですね...。
順番逆にします。

maedana commented on app/views/books/edit.html.erb:
```diff
@@ -1,6 +1,6 @@
-<h1>Editing Book</h1>
+<h1><%= t('title.Editing Book') %></h1>
```
ここに限らずですが、i18nのキーはスネークケースで書くのが一般的です。他は指摘しませんが、それぞれ直しましょうー

maedana commented on config/application.rb:
```diff
@@ -18,5 +18,6 @@ class Application < Rails::Application
     #
     # config.time_zone = "Central Time (US & Canada)"
     # config.eager_load_paths << Rails.root.join("extras")
+    # config.i18n.default_locale = :ja
```
細かいですが、いらない差分かなーと思いました。たぶん最初はこっちのファイルでdefault_locale設定しようとしたのかなって思いました。

naomichi-h replied:
> たぶん最初はこっちのファイルでdefault_locale設定しようとしたのかなって思いました。

その通りです...😅コメントアウトしていたのをすっかり忘れていました。

naomichi-h committed:
clean:レビューを受けて修正を行いました

Gemfile:gem i18n_generatorsをdevelopment groupに定義しました。
application_controller.rb:mergeメソッドの呼び出し元と引数の順番を逆にしました。
i18nのキー名の修正:スネークケースで統一しました。
余計なコメントアウトを削除しました。

naomichi-h commented:
指摘された点を修正しましたので、確認のほど、よろしくお願いいたします。

maedana commented on app/controllers/books_controller.rb:
```diff
@@ -29,7 +29,7 @@ def create
     respond_to do |format|
       if @book.save
-        format.html { redirect_to @book, notice: t('message.Book was successfully created') }
+        format.html { redirect_to @book, notice: t('message.Book_was_successfully_created') }
```
先頭が普通は小文字かなーと思います。それとキーを文章にするのはあまり見かけたことないですね。
なんでもいいんですが、対象を示すようなキーをつけることが多いかなって思います。

標準のエラーメッセージだとこんな感じですしね。
```yml
  errors:
    format: "%{attribute} %{message}"
    messages:
      accepted: must be accepted
      blank: can't be blank
      ...
```

naomichi-h replied:
なるほど...こうしてみるとかなり違和感ありますね。
修正します！

naomichi-h committed:
clean:レビューを受けて修正を行いました２

i18nのキー名の修正:頭文字を小文字にし、冗長なキーを簡潔にしました。

naomichi-h commented:
指摘された点を修正しましたので、確認のほど、よろしくお願いいたします。
